src/_pytest/skipping.py

```diff
@@ -294,3 +296,16 @@ def pytest_report_teststatus(report: BaseReport) -> Optional[Tuple[str, str, str
         elif report.passed:
             return "xpassed", "X", "XPASS"
     return None
+
+
+current_item_var: ContextVar[Item] = ContextVar("current_item_var")
+
+
+def pytest_runtest_protocol(item: Item, nextitem: Optional[Item]) -> None:
+    current_item_var.set(item)
```

Comment:
This is completely missing cleanup and self protection, it's basically breaking pytester in a hidden way